src/ConfigManager.cpp

```diff
@@ -342,6 +339,53 @@ void ConfigManager::createBaseWebServer() {
     server->onNotFound(std::bind(&ConfigManager::handleNotFound, this));
 }
 
+
+void ConfigManager::clearWifiSettings(bool reboot) {
+    char ssid[SSID_LENGTH];
+    char password[PASSWORD_LENGTH];
+    memset(ssid, NULL, SSID_LENGTH);
+    memset(password, NULL, PASSWORD_LENGTH);
+
+    DebugPrintln('Clearing WIFI connection.');
+    storeWifiSettings(magicBytesEmpty, ssid, password);
+
+    if (reboot) {
+        ESP.restart();
+    }
+}
+
+void ConfigManager::storeWifiSettings(String magicBytes, String ssid, String password) {
```

Comment:
However I passed `magicBytes` in as a char, it would not save properly, and the magicByte compare at initialization would fail and revert to AP -- I gave up, setting to `String` and converting was the only thing that worked .. thoughts?

```
(char[WIFI_OFFSET] magicBytes, String ssid, String password)
( const char* magicBytes ... 
```

Reply:
What about making it a `const char *magic`?

Reply:
tried that .. wouldnt save ... let me try your exact suggesstion ...

Reply:
Will need to look it up quick.

Reply:
ah, actually, try `char magicBytes[WIFI_OFFSET]`.

Reply:
Nope .... something doesnt write correct

```
void ConfigManager::storeWifiSettings(const char *magic, String ssid, String password) {
    //char magicBytesChar[MAGIC_LENGTH];
    char ssidChar[SSID_LENGTH];
    char passwordChar[PASSWORD_LENGTH];

    //strncpy(magicBytesChar, magicBytes.c_str(), MAGIC_LENGTH);
    strncpy(ssidChar, ssid.c_str(), SSID_LENGTH);
    strncpy(passwordChar, password.c_str(), PASSWORD_LENGTH);

    DebugPrint("Storing WIFI: ");
    DebugPrintln(ssid);

    EEPROM.put(0, magic);
```
```
MagicBytes mismatch - SSID: asdasd
```

Reply:
Its SUPER bizzaree .. and really confusing

Reply:
trying `char magicBytes[MAGIC_LENGTH]` .. think I had .. one sec

Reply:
```
error: invalid conversion from 'const char*' to 'char*' [-fpermissive]
     storeWifiSettings(magicBytes, ssid, password);
```

Reply:
ah, that kinda makes sense. `magicBytes` is a `const char magicBytes[2]` so perhaps match that.

Reply:
will try, but I think I tried that too .. ;) --- -one sec

Reply:
C++ typing is a true pain 😄

Reply:
Super baffling .... fail .. 

```
void ConfigManager::storeWifiSettings(const char magicBytes[2], String ssid, String password) {
    //char magicBytesChar[MAGIC_LENGTH];
    char ssidChar[SSID_LENGTH];
    char passwordChar[PASSWORD_LENGTH];

    //strncpy(magicBytesChar, magicBytes.c_str(), MAGIC_LENGTH);
    strncpy(ssidChar, ssid.c_str(), SSID_LENGTH);
    strncpy(passwordChar, password.c_str(), PASSWORD_LENGTH);

    DebugPrint("Storing WIFI: ");
    DebugPrintln(ssid);

    EEPROM.put(0, magicBytes);
    EEPROM.put(MAGIC_LENGTH, ssidChar);
    EEPROM.put(MAGIC_LENGTH + SSID_LENGTH, passwordChar);
    EEPROM.commit();
}
```

```
12:33:05.764 -> Storing WIFI: 12
12:33:05.946 -> Mo
12:33:05.983 -> ,⸮
12:33:05.983 -> Reading saved configuration
12:33:05.983 -> MagicBytes mismatch - SSID: 12
12:33:05.983 -> Starting Access Point
12:33:06.486 -> AP Name: Demo-
12:33:06.486 -> AP IP address: 192.168.1.1

```

Reply:
Im chaulking it up to a constant that is then jammed into a memory block ... and not understanding the mysteries

Reply:
Yeah, I think you are correct. Then I think I would prefer the `bool` over a `String` cause of the copying. WDYT?

Reply:
.. LOL .. so back to my original passing of the `reset` flag? ... I don't so much mind how it is right now .. its your call I can change in 5 minutes

Reply:
although, the way it is written leaves a little `magic` (ppun intended) to the reason for the code

Reply:
Yeah, I think back to the `reset` flag. Sorry.

Reply:
reverted .... wow .. that was a sucky moment in time

Reply:
So sorry for the fuss man.

Reply:
no worries, strong believer in code wholesomeness ... good standards... and you are the primary person that would need to clean up after me ... I totally get it ..